converter.go

```diff
@@ -375,23 +386,27 @@ func arrayToString(v driver.Value, tsmode snowflakeType, params map[string]*stri
 		}
 		if len(barr) == 0 {
 			res := "[]"
-			return bindingValue{&res, "json", &schemaForBytes}, nil
+			return bindingValue{&res, jsonFormatStr, &schemaForBytes}, nil
 		}
 		res := "["
 		for _, b := range barr {
 			res += fmt.Sprint(b) + ","
 		}
 		res = res[0:len(res)-1] + "]"
-		return bindingValue{&res, "json", &schemaForBytes}, nil
+		return bindingValue{&res, jsonFormatStr, &schemaForBytes}, nil
+	} else if stringer, ok := v1.Interface().(fmt.Stringer); ok && v1.Kind() == reflect.Array && v1.Type().Elem().Kind() == reflect.Uint8 && v1.Len() == 16 {
```

Comment:
I don't like such special cases. It may fail depending on the type defined in the table (varchar vs array vs array(byte) vs binary). Can you add such integration tests?

Reply:
@sfc-gh-pfus What's the best approach for adding integration tests? I cannot run those locally, unless your team has a method for running them

Reply:
This is testing for a very specific type of `[16]byte`

Reply:
I believe that you have to use your snowflake account to run integration test, or at least use a [trial](https://signup.snowflake.com/). I believe that if you want to add an integration test, you would have to write a test, which creates a table, performs an insert and performs a select.

Yes, this is something I'm afraid of. Let's imagine you have a UUID (which technically is an array of bytes) and you want to insert it into varchar table. With your solution you would use it like this (guessing without integration test):
```
myUuid := NewUUID()
db.Exec("INSERT ... VALUES (?)", myUuid)
```
Am I right? In that case, the actual value goes to backend as a string representation (eg `646313f9-d353-41bc-8ef4-21482bab7020`) and it is properly inserted into varchar column. So far so good.

But now, let's suppose, the column type is not varchar, but ARRAY or BINARY. From the driver perspective, you still send use the same construct:
```
myUuid := NewUUID()
db.Exec("INSERT ... VALUES (?)", myUuid)
```
But if we send it as the same value (`646313f9-d353-41bc-8ef4-21482bab7020`) backend won't be able to interpret it as array (expects `[16,84,21,47,77...]`) nor binary (expects `646313f9d35341bc8ef421482bab7020`).

But I may be wrong and maybe backend will be able to somehow magically parse each option, this is why I suggested adding integration test for all cases :) 

Generally UUID is not something that is supported by this driver (if it worked before it was only by accident) and if we want to add support for it, we should do it globally, not only in this case. Also, we have to think globally - how other drivers should support similar cases.

One more note - 16 elem byte array is not reserved for uuids, it can be also some checksum, something connected to cryptography (keys, IVs etc) and so on.

To solve your particular case - I believe, that if you have a stringer, you can replace `db.Exec("INSERT ...", u)` with `db.Exec("INSERT ..., u.String())`, right? Your option is nicer and I really would like it, if only it wasn't so troublesome :(

Reply:
UUIDs on other drivers (SQLite, Postgres, etc.) is already supported.

Right; this issue more is intrinsic to using an ORM on top of the snowflake driver. Because we cannot control the ORM's behavior how it's going to do the final insert statement, we rely on the driver to do the correct thing. I can remove the UUID check in such a way OR I can add additional logic checks that looks for specific methods being available such as Parse,ParseUUID etc.

Reply:
As far as I know postgres has support for UUID type, whereas in snowflake there is not such a special type :) You can insert uuid to varchar, array or binary columns.

Ok, ORMs are interesting topic. Do you have any particular one in mind? Is there any ORM binding to snowflake which properly generates snowflake query syntax?

My suggestion for now is to either:
1. We can think internally about this topic and present more requirements of how to support UUIDs and you can implement more complete solution :) As I mentioned - it should work for more than a varchar probably, we would have to think if it should be covered in structured types etc. And it should obviously work with reading as well. On your side - your input from ORM use cases would also be valuable in that area, since I don't have any knowledge if gosnowflake works with any ORMs. This is probably faster way.
2. You can work with you SAM/SE or file a feature request via Salesforce and we can plan it for the future. Unfortunately, this way may be much longer, I can't make promises when we can work on it. Priorities are managed mainly by product management, not devs, and there are key factors taken into account like performance, security, number of customers interested etc.

Reply:
I will work on a more complete solution

This behavior was introduced in version 1.11.0 by #1154 

Prior to that we never ran into this issue. Most people would write UUIDs as strings to their DB I would think.

We are using GORM in tandem with https://github.com/gorm-snowflake/gorm-snowflake. This is a repository we manage.

Reply:
@sfc-gh-pfus when you say read/write integration tests, do you mean the structured types tests for read/write or something else?

I also made it so we're abso-positively sure that it's a UUID implementer of RFC 4122. Are you good with this?

Reply:
I mean firstly tests in which you create a table, insert uuid into varchar column and then read it into uuid value in client code again. driver_test.go seems to be a good place for this. Secondly - yes, we should extend structured types tests. This is slightly optional, but if it is possible to add it, it would be great.

Checking your implementation of RFC - it is a great idea! This way we can have only varchar-supported :) So summing things up - your writing part seem mostly fine so far, now let's ensure that reading such UUIDs also work. Good job!

Reply:
Are we OK if I test using this?

https://snowflake.localstack.cloud/introduction/

My team prefers I dont use our real Snowflake database 😄